src/value.jl

```diff
@@ -179,3 +179,9 @@ DataAPI.defaultarray(::Type{CategoricalValue{T, R}}, N) where {T, R} =
   CategoricalArray{T, N, R}
 DataAPI.defaultarray(::Type{Union{CategoricalValue{T, R}, Missing}}, N) where {T, R} =
   CategoricalArray{Union{T, Missing}, N, R}
+
+# define appropriate handlers for JSON3 interface
+StructTypes.StructType(x::CategoricalValue) = StructTypes.StructType(get(x))
```

Comment:
You shouldn't need this definition; StructTypes.jl has a fallback like `StructTypes.StructType(x::T) where {T} = StructTypes.StructType(T)`

Reply:
But `get` makes a difference?

Reply:
Sorry, I'm saying the StructTypes.jl fallback will call `StructType` on the type of `x`, and in this PR, the next line definition "unwraps" the eltype of `CategoricalValue{T}`.

Reply:
I added this definition during testing because of some issues where the small union type was lost without it. This may be related to the stuff I got rid of in the last commit, I can't remember. I'll test it out and see if it's necessary anymore.

Reply:
Without this definition, the following error occurs. 

```julia
julia> x = CategoricalArray(["x",nothing,"y","z","y",nothing,"z","x"])
julia> str = JSON3.write(x)

ERROR: ArgumentError: CategoricalPool{Union{Nothing, String},UInt32,CategoricalValue{Union{Nothing, String},UInt32}} doesn't have a defined `StructTypes.StructType`
Stacktrace:
 [1] write(::StructTypes.NoStructType, ::Array{UInt8,1}, ::Int64, ::Int64, ::CategoricalPool{Union{Nothing, String},UInt32,CategoricalValue{Union{Nothing, String},UInt32}}; kw::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/write.jl:69
 [2] write(::StructTypes.NoStructType, ::Array{UInt8,1}, ::Int64, ::Int64, ::CategoricalPool{Union{Nothing, String},UInt32,CategoricalValue{Union{Nothing, String},UInt32}}) at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/write.jl:69
 [3] #_#47 at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/write.jl:89 [inlined]
 [4] WriteClosure at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/write.jl:80 [inlined]
 [5] foreachfield at /Users/davidlittle/.julia/packages/StructTypes/CpLmq/src/StructTypes.jl:568 [inlined]
 [6] #write#48 at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/write.jl:100 [inlined]
 [7] write at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/write.jl:98 [inlined]
 [8] write(::StructTypes.ArrayType, ::Array{UInt8,1}, ::Int64, ::Int64, ::CategoricalArray{Union{Nothing, String},1,UInt32,Union{Nothing, String},CategoricalValue{Union{Nothing, String},UInt32},Union{}}; kw::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/write.jl:132
 [9] write at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/write.jl:128 [inlined]
 [10] write(::CategoricalArray{Union{Nothing, String},1,UInt32,Union{Nothing, String},CategoricalValue{Union{Nothing, String},UInt32},Union{}}; kw::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/write.jl:24
 [11] write(::CategoricalArray{Union{Nothing, String},1,UInt32,Union{Nothing, String},CategoricalValue{Union{Nothing, String},UInt32},Union{}}) at /Users/davidlittle/.julia/packages/JSON3/pjUse/src/write.jl:22
 [12] top-level scope at REPL[46]:1
 [13] eval(::Module, ::Any) at ./boot.jl:331
 [14] eval_user_input(::Any, ::REPL.REPLBackend) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [15] run_backend(::REPL.REPLBackend) at /Users/davidlittle/.julia/packages/Revise/BqeJF/src/Revise.jl:1184
 [16] top-level scope at none:0
```